pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/PulsarAdminTool.java

```diff
@@ -49,7 +48,7 @@ public class PulsarAdminTool {
 
     private static int lastExitCode = Integer.MIN_VALUE;
 
-    protected List<CustomCommandFactory> customCommandFactories = new ArrayList();
+    protected static List<CustomCommandFactory> customCommandFactories;
```

Comment:
In pulsar shell a new instance of PulsarAdminTool (actually AdminShell) is created when the current config changes.
I think it's fine to just cache it locally to the instance and remove `static`